test/SecTester.Scan.Tests/ScanTests.cs

```diff
@@ -1,53 +1,228 @@
+
+    // assert
+    scan.Active.Should().BeTrue();
+  }
+
+  [Fact]
+  public async Task Done_DefaultValue_ReturnsTrue()
+  {
+    // act
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions());
+
+    // assert
+    scan.Done.Should().BeFalse();
+  }
+
+  [Fact]
+  public async Task Status_ReturnsState()
   {
     // arrange
-    var scans = Substitute.For<Scans>();
+    var scanState = new ScanState(ScanStatus.Running);
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    _scans.GetScan(ScanId).Returns(Task.FromResult(scanState));
 
     // act
-    var act = () => new Scan(scans, null!);
+    var result = await scan.Status().FirstAsync();
 
     // assert
-    act.Should().Throw<ArgumentNullException>().WithMessage("*options*");
+    result.Should().Be(scanState);
   }
 
   [Fact]
-  public void Active_DefaultValue_ReturnTrue()
+  public async Task Status_GivenCancelledToken_ThrowError()
   {
     // arrange
-    var scans = Substitute.For<Scans>();
+    using var cts = new CancellationTokenSource();
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    cts.Cancel();
 
     // act
-    using var scan = new Scan(scans, new ScanOptions());
+    var act = async () => await scan.Status(cts.Token).FirstAsync();
 
     // assert
-    scan.Active.Should().BeTrue();
+    await act.Should().ThrowAsync<OperationCanceledException>();
   }
 
   [Fact]
-  public void Done_DefaultValue_ReturnTrue()
+  public async Task Status_SwitchingDoneState_StopsYield()
   {
     // arrange
-    var scans = Substitute.For<Scans>();
+    var runningScanState = new ScanState(ScanStatus.Running);
+    var stoppedScanState = new ScanState(ScanStatus.Stopped);
+
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    _scans.GetScan(ScanId).Returns(Task.FromResult(runningScanState),
+      Task.FromResult(stoppedScanState));
 
     // act
-    using var scan = new Scan(scans, new ScanOptions());
+    var result = await scan.Status().LastOrDefaultAsync();
 
     // assert
-    scan.Done.Should().BeFalse();
+    result.Should().BeEquivalentTo(stoppedScanState);
+  }
+
+  [Fact]
+  public async Task Status_GivenQueuedStateConsequently_LogsConcurrencyWarning()
+  {
+    // arrange
+    var scanState = new ScanState(ScanStatus.Queued);
+
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions(PollingInterval: TimeSpan.Zero));
+
+    _scans.GetScan(ScanId).Returns(Task.FromResult(scanState),
+      Task.FromResult(scanState));
+
+    // act
+    await scan.Status().Take(2).LastOrDefaultAsync();
+
+    // assert
+    _logger.Received(1).Log(LogLevel.Warning, Arg.Is<string>(s => s.Contains("increase the execution concurrency")));
+  }
+
+  [Fact]
+  public async Task Status_GivenAnyStateAfterQueued_LogsResumingExecution()
```

Comment:
Please cover the case when `Status` stops consuming while receiving the done-like statuses

Reply:
`Status_SwitchingDoneState_StopsYield` is responsible for this
https://github.com/NeuraLegion/sectester-net/pull/59#discussion_r1035229969

Reply:
This test covers only a single case with `done` when you have two more states that lead to the same behavior. BTW never a close conversation which requires my attention.